### DISCUSSION TITLE
Getting the newest version of standard libraries (e.g. "Plot")

### DISCUSSION THREAD

**Discussion by ratnanil** [quarto-dev/quarto-cli#3677] [Q&A] [answered]
Hi! I'd like to work with the newest version of the standard library `Plot`. If I include the following `ojs` chunk in my notebook, I get back the version `0.6.0`. However, I need version `0.6.1`. I do have the newest quarto version (`1.2.280`), 

````
```{ojs}
Plot.version
```
````

Do I have to wait for quarto to update the standard library, or is there a way that I import a newer version of Plot locally?

**Comment by mcanouil** [comment]
`1.2.280` is the latest stable version, not the latest.
Within a stable release, it's unlikely you'll get a library upgrade.
You can try the pre-release version, or embed observable yourself in your Quarto document using the `include-*` options.

**Comment by cscheid** [comment]
@ratnanil:

> If I include the following ojs chunk in my notebook, I get back the version 0.6.0. However, I need version 0.6.1

We try to update the OJS standard libraries at every release. 0.6.1 was released 3 days ago. It's not practical for us to track them this closely given the rest of the things we want to do in quarto.

> Do I have to wait for quarto to update the standard library, or is there a way that I import a newer version of Plot locally?

Try `https://cdn.skypack.dev/@observablehq/plot`

@mcanouil:

> You can try the pre-release version, or embed observable yourself in your Quarto document using the include-* options.

Note that embedding _observable_ is not really a practical choice in Quarto, because you'll need the full runtime, a custom parser, etc.

**Comment by ratnanil** [comment]
> Try https://cdn.skypack.dev/@observablehq/plot

Like so?

```js
import { Plot } from "https://cdn.skypack.dev/@observablehq/plot"
```

(this does not work)

**Reply by cscheid** [reply]
There could be two reasons. There might be an issue with name clashes:

```js
import { Plot as NewPlot } from "https://cdn.skypack.dev/@observablehq/plot"
```

or you might need a default import:

```js
import NewPlot from "https://cdn.skypack.dev/@observablehq/plot"
```

**Reply by ratnanil** [reply]
Neither works...

> `import { Plot as NewPlot } from "https://cdn.skypack.dev/@observablehq/plot"`

This throws the following error: `TypeError: error loading dynamically imported module`

> `import NewPlot from "https://cdn.skypack.dev/@observablehq/plot"`

This throws the following error: `OJS Syntax Error (line 7, column 8)Unexpected token ` (see below)

![image](https://user-images.githubusercontent.com/12532091/208111577-b8eecb84-9872-41f2-a101-caca963bc0d3.png)

**Reply by cscheid** [reply, marked as answer]
Sorry, I always forget that the standard syntax for imports doesn't work for observable. In addition, skypack is currently broken for d3 v7.7.0, which Plot uses. Use this instead:

````
```{ojs}
NewPlot = import("https://esm.sh/@observablehq/plot") 
NewPlot
```
````

<img width="863" alt="image" src="https://user-images.githubusercontent.com/285675/208118532-a88f607d-9d62-4e28-a2da-89861235714f.png">

**Reply by ratnanil** [reply]
Perfect, this works! I can even overwrite Plot, so the following line is fine. `Plot.version` now returns undefined for some reason, but I can confirm it's the newest version `0.6.1` because the functionality I needed is now available. Thanks so much!

```js
Plot = import("https://esm.sh/@observablehq/plot") 
```

**Reply by cscheid** [reply]
Good to know. FWIW, the `version` field is put there by ObservableHQ's runtime, and that's why you don't see it in a plain JS import.

**Comment by mcanouil** [comment]
FYI, I made a PR to document this:

- https://github.com/quarto-dev/quarto-web/pull/802
